tidy3d/plugins/design/method.py

```diff
@@ -302,7 +681,13 @@ class MethodRandomCustom(AbstractMethodRandom):
         " can also be supplied.",
     )
 
-    @pd.validator("sampler")
+    sampler_kwargs: Any = pd.Field(
+        {},
```

Comment:
hm, so dicts are mutable so this seems a bit unusual to me. What's the idea here? Also not sure about the hasattr business below.

Ideal: let's just remove the `MethodRandom`? or just have it be a simple class that inherits from `MethodMonteCarlo` with a warning (through a `pd.root_validator` if it's initialized?

Reply:
Yeah we could just remove MethodRandom as the user can reimplement it in MethodRandomCustom come to think of it.

For the code above it's a dict because we needed to initialise the sampler in 'get_sampler' (so that random seed is set properly) and needed to pass the kwargs for the sampler through to this stage. Is there another way of doing this?

Reply:
I suppose this works. it just begs the question why not have it this way for eg the GA code? (pygad kwargs)? I think the idea is that having things listed out as fields makes things better documented, easier to read and write to file, and we can catch things at the validation stage before they cause errors.

Are the sampler kwargs the same over all of the possible random custom samplers?  would it make sense to just remove Custom Random samplers too and only support the builtin Monte Carlo ones?